tests/unit/multi_tenancy.cpp

```diff
@@ -0,0 +1,390 @@
+// Copyright 2024 Memgraph Ltd.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt; by using this file, you agree to be bound by the terms of the Business Source
+// License, and you may not use this file except in compliance with the Business Source License.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+#include <algorithm>
+#include <cstdlib>
+#include <filesystem>
+#include <thread>
+
+#include "communication/bolt/v1/value.hpp"
+#include "communication/result_stream_faker.hpp"
+#include "csv/parsing.hpp"
+#include "dbms/dbms_handler.hpp"
+#include "disk_test_utils.hpp"
+#include "flags/run_time_configurable.hpp"
+#include "glue/communication.hpp"
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+#include "interpreter_faker.hpp"
+#include "license/license.hpp"
+#include "query/auth_checker.hpp"
+#include "query/config.hpp"
+#include "query/exceptions.hpp"
+#include "query/interpreter.hpp"
+#include "query/interpreter_context.hpp"
+#include "query/metadata.hpp"
+#include "query/stream.hpp"
+#include "query/typed_value.hpp"
+#include "query_common.hpp"
+#include "replication/state.hpp"
+#include "storage/v2/inmemory/storage.hpp"
+#include "storage/v2/isolation_level.hpp"
+#include "storage/v2/property_value.hpp"
+#include "storage/v2/storage_mode.hpp"
+#include "utils/logging.hpp"
+#include "utils/lru_cache.hpp"
+#include "utils/synchronized.hpp"
+
+namespace {
+std::set<std::string> GetDirs(auto path) {
+  std::set<std::string> dirs;
+  // Clean the unused directories
+  for (const auto &entry : std::filesystem::directory_iterator(path)) {
+    const auto &name = entry.path().filename().string();
+    if (entry.is_directory() && !name.empty() && name.front() != '.') {
+      dirs.emplace(name);
+    }
+  }
+  return dirs;
```

Comment:
Can we isolate this function into header file, it seems duplicated as I have seen it on one more place

Reply:
I'll just leave them as is. 
They are used in two different tests and we don't have a tests common directory of library.